### ISSUE TITLE
Whitespace trimming on anchor tags (when using svg tag)

### ISSUE THREAD

**Issue by mm98** [Taritsyn/WebMarkupMin#154]
Hi,

It looks like there has been added a bit too much whitespace trimming in version `2.13.6`. This worked correctly in version `2.13.5`. It is only happening before and after an `anchor` tag. I've noticed it on `string.Format()` and normal inline `A` tags.

`We've located you <a href="{0}" target="_blank">here</a> with an inaccuracy of {1} km.`

Version `2.13.5`
![image](https://user-images.githubusercontent.com/992714/215250448-13255542-4deb-4e13-bd75-91085fce7fc3.png)
![image](https://user-images.githubusercontent.com/992714/215250776-f94eebd4-74ce-42c6-8e12-f13721c83f45.png)

Version `2.13.6`
![image](https://user-images.githubusercontent.com/992714/215250573-4d4c63ef-07be-4d95-858b-3299c55b8931.png)
![image](https://user-images.githubusercontent.com/992714/215250749-fae6943d-f75b-4f43-a495-d529f3eec9b1.png)

WebMarkup configuration in startup.cs:
```csharp
services
	.AddWebMarkupMin(options =>
	{
		options.AllowMinificationInDevelopmentEnvironment = true;
		options.DisablePoweredByHttpHeaders = true;
	})
	.AddHtmlMinification(options =>
	{
		options.MinificationSettings.RemoveEmptyAttributes = true;
		options.MinificationSettings.RemoveJsTypeAttributes = true;
		options.MinificationSettings.RemoveOptionalEndTags = true;
		options.MinificationSettings.RemoveRedundantAttributes = true;
		options.MinificationSettings.RemoveTagsWithoutContent = false;
		options.MinificationSettings.WhitespaceMinificationMode = WhitespaceMinificationMode.Medium;
		options.SupportedHttpMethods = new HashSet<string>(StringComparer.OrdinalIgnoreCase) { "GET", "POST" };
	});
```

I have tried to remove `WhitespaceMinificationMode = WhitespaceMinificationMode.Medium;`, the result is still the same.

**Comment by Taritsyn** [comment]
Hello!

I can't reproduce this error. You can check it yourself on the [WebMarkupMin Online](https://webmarkupmin.bsite.net/minifiers/html-minifier) site.

Maybe problem is in CSS styles? In any case, you need to look at the input and output HTML code.

**Comment by Taritsyn** [comment]
Are conditional comments used in your code?

**Comment by mm98** [comment]
It looks like this is caused by the `<svg><use ... ></use></svg>` tags. This will strip whitespace in any following or nested elements (I think). There are no conditional comments defined in the code.

Source (cshtml):
```html
<div>
	<tabler icon="World" />
	<a href="#test1">Danish</a> |
	<a href="#test1">English</a>
</div>
```
Note `<tabler />` is a TagHelper rendering `<svg><use ... ></use></svg>` tags.

Rendered without using WebMarkup:
```html
<div>
	<svg class="icon-svg" aria-label="World icon"><use href="/img/icons-tabler.svg#world"></use></svg>
	<a href="#test1">Danish</a> |
	<a href="#test1">English</a>
</div>
```

Rendered with WebMarkup v2.13.5 (correct - whitespace is maintained):
```html
<div><svg class="icon-svg" aria-label="World icon"><use href="/img/icons-tabler.svg#world"></use></svg> <a href="#test1">Danish</a> | <a href="#test1">English</a></div>
```

Rendered with WebMarkup v2.13.6 (incorrect - all whitespace has been removed):
```html
<div><svg class="icon-svg" aria-label="World icon"><use href="/img/icons-tabler.svg#world"></use></svg><a href="#test1">Danish</a>|<a href="#test1">English</a></div>
```

## Reproduce 

This can also be reproduced in [WebMarkupMin Online](https://webmarkupmin.bsite.net/minifiers/html-minifier) with the following HTML:

```html
<div><svg class="icon-svg" aria-label="World icon"><use href="/img/icons-tabler.svg#world"></use></svg> <a href="#test1">Danish</a> | <a href="#test1">English</a></div>
```

**Comment by Taritsyn** [comment]
Thank you for the information! In the latest version, I changed the logic of processing XML-based tags (such as `svg`).

I'll try to release a fix tonight.

**Comment by Taritsyn** [comment]
This error has been fixed in [version 2.13.7](https://github.com/Taritsyn/WebMarkupMin/releases/tag/v2.13.7).

**Comment by mm98** [comment]
I can confirm that this is fixed. Thanks you so much!

**Comment by dotnetshadow** [comment]
I have the same issue with ```<b>``` and ```<strong>``` tags, it's a very similar situation
![image](https://user-images.githubusercontent.com/296247/217758403-07615e64-26f5-4edd-8b6a-966a44fcfd0b.png)

```html
<p class="alert mt-4" style="border:1px solid #d3d6de;background:#fff"><svg class="svg-inline--fa fa-fw" role="presentation"><use href="#far-fa-clock"></use></svg>Trading takes place from<strong>10:00 AM</strong>to<strong>4:00 PM</strong>,<strong>AEDT</strong></p>
```

**Comment by Taritsyn** [comment]
@DotnetShadow, have you tried updating to [version 2.13.7](https://github.com/Taritsyn/WebMarkupMin/releases/tag/v2.13.7)?

**Comment by dotnetshadow** [comment]
> @dotnetshadow, have you tried updating to [version 2.13.7](https://github.com/Taritsyn/WebMarkupMin/releases/tag/v2.13.7)?

Thanks for the quick reply. Yep, that's the version I'm using

In my razor file I have the following:
```razor
<p class="alert mt-4" style="border: 1px solid #d3d6de; background: #fff;">
    <svg-icon Name="far-fa-clock" />
    <span>Trading takes place from <strong>@DateTime.Today.Add(Model.OpenTime).ToString("h:mm tt")</strong> to <strong>@DateTime.Today.Add(Model.CloseTime).ToString("h:mm tt")</strong>, <strong>@Model.TimezoneAbbreviation</strong></span>
</p>
```

`<svg-icon>` is my tag helper
When I disable WebMarkupMin it works correctly

**Comment by Taritsyn** [comment]
The following HTML code:

```html
<p class="alert mt-4" style="border: 1px solid #d3d6de; background: #fff;">
    <svg class="svg-inline--fa fa-fw" role="presentation"><use href="#far-fa-clock"></use></svg>
    <span>Trading takes place from <strong>10:00 AM</strong> to <strong>4:00 PM</strong>, <strong>AEDT</strong></span>
</p>
```

after processing by the HTML minifier version 2.13.7, in my opinion, looks absolutely correct:

```html
<p class="alert mt-4" style="border:1px solid #d3d6de;background:#fff"><svg class="svg-inline--fa fa-fw" role=presentation><use href=#far-fa-clock></use></svg> <span>Trading takes place from <strong>10:00 AM</strong> to <strong>4:00 PM</strong>, <strong>AEDT</strong></span></p>
```

I recommend that you check again a versions of NuGet packages you are using.

**Comment by dotnetshadow** [comment]
Yeah I'll take another look but I think I have it correctly installed
![image](https://user-images.githubusercontent.com/296247/217791543-911cc8b4-68f0-4e4a-996d-3e4c15171f5f.png)

In the browser this is how the markup looks
![image](https://user-images.githubusercontent.com/296247/217792151-94ae3cc0-73d2-470a-9cc1-9005dc21742e.png)

![image](https://user-images.githubusercontent.com/296247/217792284-8b586289-9406-4314-be72-67fc44f46362.png)

**Comment by dotnetshadow** [comment]
Maybe we can setup a teams meeting or something to try and debug it?

**Comment by Taritsyn** [comment]
This may be a deployment error in which old assemblies are not overwritten.

If everything is fine with the assembly versions, then you need try to cover a larger amount of HTML code. For testing, you can use the [WebMarkupMin Online](https://webmarkupmin.bsite.net/minifiers/html-minifier) site.

**Comment by dotnetshadow** [comment]
I did check while developing and enabled the option and it still produced the same result. 
I cleaned out the obj and bin folders... for now I have to insert `&nbsp;`

**Comment by Taritsyn** [comment]
Show me the full contents of the `p` tag as text.

**Comment by dotnetshadow** [comment]
Actually I did another test
```<p>Hello <span>test</span></p>```

and it outputs with a problem too

![image](https://user-images.githubusercontent.com/296247/217796035-9ff9fc9f-e033-4ab9-9919-dfc0813ae5fc.png)

Maybe something else going on?

**Comment by Taritsyn** [comment]
> I cleaned out the obj and bin folders... for now I have to insert &nbsp;

In such situations, you can always use a [ignoring comment tags](https://github.com/Taritsyn/WebMarkupMin/wiki/Ignoring-fragments-of-markup), but I would like to get an example of a real HTML code leading to an error.

**Comment by dotnetshadow** [comment]
You can use this link, to see the problem
https://preview.stockwatch.com.au/market/hours

**Comment by Taritsyn** [comment]
> and it outputs with a problem too

Show me an example of a full HTML document in the form of text.

**Comment by dotnetshadow** [comment]
> > and it outputs with a problem too
> 
> Show me an example of a full HTML document in the form of text.

You mean like a static html page?

**Comment by dotnetshadow** [comment]
Any way I can message you directly?

**Comment by Taritsyn** [comment]
For now, this example is enough.

**Comment by dotnetshadow** [comment]
> For now, this example is enough.

So the link I provided is enough to test with?

**Comment by dotnetshadow** [comment]
For the following test case  `<p>Hello <span>test</span></p>`, 
2.13.5 - WORKS
2.13.6 - NOT WORKING
2.13.7 - NOT WORKING
2.13.6 and 2.13.7 will only work if `options.MinificationSettings.WhitespaceMinificationMode = WhitespaceMinificationMode.None;`

**2.13.5 - WORKING**
![image](https://user-images.githubusercontent.com/296247/217811642-95626ba9-9465-4f2a-a93f-879014845e7f.png)
![image](https://user-images.githubusercontent.com/296247/217812070-88b85b7b-0551-4b5a-a5c2-9c700c11dbbb.png)

**2.13.6-2.13.7 - NOT WORKING**
![image](https://user-images.githubusercontent.com/296247/217812148-3edcb137-f773-47b6-b90f-61d8010e43d9.png)
![image](https://user-images.githubusercontent.com/296247/217812226-97d14655-9693-4878-ac13-143d71225442.png)

**UPDATE**
Seems like everything will work correctly if I use **2.13.5** including the svg case
Another way to know it's working correctly, is when I view the source for my app, the 2.13.5 shows all in 1 line, but for versions after this isn't in one line

**Comment by Taritsyn** [comment]
I have already localized the error. Now I'm thinking about how to fix it.

**Comment by dotnetshadow** [comment]
> I have already localized the error. Now I'm thinking about how to fix it.
Well done, thanks for looking into this, much appreciated

**Comment by Taritsyn** [comment]
@dotnetshadow, this error has been fixed in [version 2.13.8](https://github.com/Taritsyn/WebMarkupMin/releases/tag/v2.13.8). Thanks for providing examples of code!

**Comment by dotnetshadow** [comment]
> @dotnetshadow, this error has been fixed in [version 2.13.8](https://github.com/Taritsyn/WebMarkupMin/releases/tag/v2.13.8). Thanks for providing examples of code!

Thank you so much, I can confirm that it works correctly. Much appreciated 👍
